inc/cloudfront_media_purge/namespace.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Altis\Cloud\Cloudfront_Media_Purge;
+
+use Exception;
+use function Altis\get_aws_sdk;
+
+function bootstrap() {
+	add_action( 'delete_attachment', __NAMESPACE__ . '\\purge_media_file_cache', 100 );
+}
+
+/**
+ * Return an AWS CloudFront Client instance
+ *
+ * @return \Aws\CloudFront\CloudFrontClient
+ */
+function get_aws_client() {
+	return get_aws_sdk()->createCloudFront( [
+		'version' => 'latest',
+	] );
+}
+
+/**
+ * Purge an attachment file cache from CloudFront
+ *
+ * It sends a request to invalidate all sizes given for a media file
+ *
+ * @param int $media_id Attachment ID.
+ *
+ * @return bool
+ */
+function purge_media_file_cache( $media_id ) {
+	$meta = wp_get_attachment_metadata( $media_id );
+	if ( ! $meta ) {
+		return false;
+	}
+	$baseurl = '/tachyon/';
+	$basedir = trailingslashit( dirname( $meta['file'] ) );
+	$items   = [ $baseurl . $meta['file'] . '*' ];
+
+	if ( ! empty( $meta['sizes'] ) ) {
+		foreach ( $meta['sizes'] as $size => $size_meta ) {
+			$items[] = $baseurl . $basedir . $size_meta['file'] . '*';
+		}
+	}
+
+	$client = get_aws_client();
+
+	$distribution_id = apply_filters( 'altis_cloudfront_media_purge_distribution_id', defined( 'CLOUDFRONT_DISTRIBUTION_ID' ) ? CLOUDFRONT_DISTRIBUTION_ID : '' );
+
+	if ( empty( $distribution_id ) ) {
+		trigger_error( 'Empty cloudfront distribution id for media purge request.', E_USER_WARNING );
+		return false;
+	}
+
+	try {
+		$client->createInvalidation( [
+			'DistributionId'    => $distribution_id,
+			'InvalidationBatch' => [
+				'CallerReference' => current_time( 'timestamp' ),
```

Comment:
Think you could just use `time()` here - does this property need to be unique? Is it actually date related?

It's unlikely but if it does need to be unique I'd incorporate the site ID and media ID perhaps.

Reply:
I think this will collide easily too, as concurrent processes or even a loop running coud easily create more than one per second. I'd use something including rand()

Reply:
Rand also will not work as expected -  as per doc 
```
If CallerReference is a value you already sent in a previous invalidation batch request but the content of any Path is different from the original request, CloudFront returns an InvalidationBatchAlreadyExists error.
```

I am not able to find any character limit so Either I can go with `blog_id-media_id` or `md5( wp_json_encode( $paths_patterns ) )`